packages/common/src/compiler/filtersCompiler.ts

```diff
@@ -46,9 +46,9 @@ export const renderStringFilterSql = (
         case FilterOperator.NOT_EQUALS:
             return !escapedFilterValues || escapedFilterValues.length === 0
                 ? 'true'
-                : `(${dimensionSql}) NOT IN (${escapedFilterValues
+                : `((${dimensionSql}) NOT IN (${escapedFilterValues
                       .map((v) => `${stringQuoteChar}${v}${stringQuoteChar}`)
-                      .join(',')})`;
+                      .join(',')} ) OR (${dimensionSql}) IS NULL)`;
```

Comment:
Do we need the same fix for numbers `renderNumberFilterSql` ?

Reply:
Added dates and numbers 

![Screenshot from 2023-05-30 12-43-14](https://github.com/lightdash/lightdash/assets/1983672/deaf285e-f5d8-4396-975e-8e8b08513913)